examples/win-guid.c

```diff
@@ -117,20 +156,48 @@ void print_os_version(vmi_instance_t vmi, addr_t kernel_base_p, uint8_t* pe) {
                 printf(" Windows 7");
         if (minor_os_version == 2)
                 printf(" Windows 8");
+    } else
+    if(major_os_version == 10) {
+        if (minor_os_version == 0)
+                printf(" Windows 10");
     } else {
-            printf("OS version unknown or not Windows\n");
+            printf(" OS version unknown or not Windows\n");
     }
 
     printf("\n");
 
 }
 
+bool kernel_debug_search(vmi_instance_t vmi, struct cv_info_pdb70 *pdb_header) {
+    addr_t i;
+    for(i=0; i < max_mem; i += PAGE_SIZE) {
+        uint8_t pe[VMI_PS_4KB];
+        vmi_read_pa(vmi, i, pe, VMI_PS_4KB);
+        uint32_t c = 0;
+        for (c;c<VMI_PS_4KB-12;c++) {
+            if(!strncmp((char*)&pe[c], "ntkrnlmp.pdb", 12) ||
+               !strncmp((char*)&pe[c], "ntoskrnl.pdb", 12) ||
+               !strncmp((char*)&pe[c], "ntkrnlpa.pdb", 12) ||
+               !strncmp((char*)&pe[c], "ntkrpamp.pdb", 12)
+            ) {
+                vmi_read_pa(vmi, i+c - 2*sizeof(uint32_t) - sizeof(struct guid), pdb_header, sizeof(struct cv_info_pdb70)+12);
```

Comment:
Well, we know what we are looking for is only 12 character long on top of the pdb_header so it's not really variable length.